File: MPI/MPI_day1_2022/processor_name.c
#include <stdio.h>
#include <mpi.h>

int main(int argc, char **argv)
{
    int size, rank;
    MPI_Init(&argc, &argv);
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
 // Get the name of the processor
    char name[MPI_MAX_PROCESSOR_NAME];
    int name_len;
    MPI_Get_processor_name(name, &name_len);
    printf("My name is %s, rank %d out of %d processors\n",name,rank,size);
    MPI_Finalize();
    return 0;
}

File: MPI/MPI_day1_2022/processor_name.f
      program main
      include 'mpif.h' 

      integer ierr, rank, size
      character*(MPI_MAX_PROCESSOR_NAME) name
      integer name_len

      call MPI_INIT(ierr)
      call MPI_COMM_SIZE(MPI_COMM_WORLD,size,ierr)
      call MPI_COMM_RANK(MPI_COMM_WORLD,rank,ierr)

      call MPI_Get_processor_name(name,name_len,ierr)
      print*,"My name is", name,"rank",rank,"out of ",size,"processors"

      call MPI_FINALIZE(ierr)
      end
